### PR TITLE
mrc-6242 Support random in generators

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Adds a random property to DiscreteSystem, which is passed to generators as parameter of `initial` and `update`. Also adds `Walk` generator for testing. 

Also deleted a couple of old POC test files, which are superseded by DiscreteSystem.test.ts. CodeCov is still grumbling though...

### CODE REVIEW

File: tests/examples/discreteWalk.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    internal(shared: WalkShared): null {
+        return null;
+    },
+
+    packingState(shared: WalkShared): Packer {
+        const shape = new Map<string, number[]>([["values", [shared.n]]]);
```
[the remark this answers is not in the document]
Comment (richfitz) [reply tests/examples/discreteWalk.ts:46]: yeah, the walk model is not terribly interesting really, and a pathalogical corner case for the concept of packing :)

File: tests/discrete.test.ts [hunk 2]
```diff
@@ -1,15 +0,0 @@
-import { expect, test } from "vitest";
-
-import { discreteSIR } from "./examples/discreteSIR";
-
-test("can do anything with our interface", () => {
```
Comment (richfitz) [inline tests/discrete.test.ts:5]: you can restore this test if we support deterministic RNG, but it looks like that never got added to the random-js package.  The idea is that we have a random object that always returns expectations, which makes these tests much easier to write

**Review by richfitz — APPROVED** [review]
LGTM - the testing of the SIR model will be easier to do if we add the deterministic generator, but we can sort that out later
